lms/djangoapps/badges/api/tests.py

```diff
@@ -22,6 +22,13 @@ class UserAssertionTestCase(UrlResetMixin, ModuleStoreTestCase, ApiTestCase):
     """
     Mixin for badge API tests.
     """
+    URLS_AUTO_RESET = False
+
+    @classmethod
+    def setUpClass(cls):
```

Comment:
Ack, yeah. Nice catch. :-) This was me punting an issue and forgetting to come back to it later. The issue was that the `override_settings` decorator is on the class as a whole, meaning that any teardown I did would have to explicitly re-override the settings to use the real ones as part of the class teardown to properly clean up. And I thought to myself, "That's kinda silly and error-prone. We should probably do something more consistent so that we always know the 'real' state of how URLs are supposed to be without overrides so we can go back to that." And then I was thinking, "Why reset/re-import URLs at all, maybe we could just have some way to flag views and have middleware that intercepts calls to them based on settings." And then I mumbled something about yaks, promised myself I'd think about it later, and promptly forgot. :-P